spring-integration-core/src/main/java/org/springframework/integration/transaction/DefaultTransactionSynchronizationFactory.java

```diff
@@ -46,13 +49,16 @@ public DefaultTransactionSynchronizationFactory(TransactionSynchronizationProces
 	@Override
 	public TransactionSynchronization create(Object key) {
 		Assert.notNull(key, "'key' must not be null");
-		DefaultTransactionalResourceSynchronization synchronization = new DefaultTransactionalResourceSynchronization(key);
-		TransactionSynchronizationManager.bindResource(key, synchronization.getResourceHolder());
-		return synchronization;
+		if (!TransactionSynchronizationManager.hasResource(key)) {
```

Comment:
??? That isn't a `TransactionSynchronizationFactory` contract. That is only our impls side effect that we register something to the TX.
The custom one can return the same `Sync` for the same `key`, or always new, or always `null`.
That is only our design that we bind resource to the TX. 
In fact (according this JIRA) we can't register into the same resource and according our logic we even don't need to create a new `Sync`. The custom `factory` can do what ever it needs: creates and returns a new instance, but bind if it is necessary for their purpose.

On the other hand renaming method (contract in the interface) doesn't protect us from the unpredictable custom implementations.
We may just say on the `create()` JavaDocs that implementation may return `null`.